src/Cmp/Queues/Infrastructure/AWS/v20121105/Queue/QueueReader.php

```diff
@@ -59,18 +61,21 @@ public function __construct(
 
     /**
      * @param callable $callback
-     * @param int      $timeout
+     * @param int $timeout
      *
      * @throws GracefulStopException
-     * @throws TimeoutReaderException
+     * @throws ParseMessageException
      * @throws ReaderException
+     * @throws TimeoutReaderException
      */
     public function read(callable $callback, $timeout=0)
     {
         $this->messageHandler->setCallback($callback);
 
         try {
             $this->consume($timeout);
+        } catch(ParseMessageException $e){
```

Comment:
Are you only adding this logic to AWS consumers? what about the others (RabbitMQ for instance) ?